catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/controller/SpriteController.java

```diff
@@ -106,10 +110,18 @@ public Sprite copyForCloneBrick(Sprite spriteToCopy) {
 
 		sprite.getSoundList().addAll(spriteToCopy.getSoundList());
 		sprite.getNfcTagList().addAll(spriteToCopy.getNfcTagList());
+		DeviceVariableAccessor accessor = new DeviceVariableAccessor(currentProject.getDirectory());
 
-		for (UserVariable userVariable : spriteToCopy.getUserVariables()) {
-			sprite.getUserVariables().add(new UserVariable(userVariable));
+		for (UserVariable originalVariable : spriteToCopy.getUserVariables()) {
+			if (checkDeviceVariableKeyEmpty(originalVariable)) {
+				accessor.addKeyToUserVariable(originalVariable);
+			}
+
+			UserVariable copyVariable = new UserVariable(originalVariable);
+			copyVariable.setDeviceValueKey(originalVariable.getDeviceValueKey());
```

Comment:
is there any time where we want to create a new user variable from an existing one without giving it the same UUID?
if not then this should be in the constructor.

Reply:
Yes, for example if a Sprite is copied, the names of their local vars should be the same, but not their keys.